src/main/java/org/junitpioneer/jupiter/ExpectedToFailExtension.java

```diff
@@ -42,7 +43,12 @@ private static void invokeAndInvertResult(Invocation<Void> invocation, Extension
 				throw t;
 			}
 
-			String message = getExpectedToFailAnnotation(extensionContext).value();
+			ExpectedToFail expectedToFail = getExpectedToFailAnnotation(extensionContext);
+			if (Stream.of(expectedToFail.onExceptions()).noneMatch(clazz -> clazz.isInstance(t))) {
+				fail("Test marked as 'expected to fail' failed with an unexpected exception", t);
```

Comment:
```suggestion
				fail("Test marked as temporarily 'expected to fail' failed with an unexpected exception", t);
```

Reply:
@knutwannheden You marked this as resolved but I don't see the change?